packages/zudoku/src/lib/ui/ActionButton.tsx

```diff
@@ -19,7 +19,7 @@ export const ActionButton = forwardRef<HTMLButtonElement, ActionButtonProps>(
             <Spinner />
           </div>
         )}
-        <div className={cn(isPending && "invisible")}>{children}</div>
+        {isPending ? <div className="invisible">{children}</div> : children}
```

Comment:
What if we used a `span` as `block` element?

```suggestion
        <span className={cn("block", isPending && "invisible")}>{children}</span>
```